python/ray/train/huggingface/transformers/_transformers_utils.py

```diff
@@ -213,3 +222,74 @@ def on_train_end(self, args, state, control, **kwargs):
             **self.delayed_report["metrics"],
         }
         self._report()
+
+
+@PublicAPI(stability="alpha")
+class RayTrainReportCallback(TrainerCallback):
+    """A simple callback to report checkpoints and metrics to Ray Tarin.
+
+    This callback overrides the `TrainerCallback.on_save()` method. After
+    a new checkpoint get saved, it fetches the latest metric dictionary
+    from `TrainerState.log_history` and reports it with the latest checkpoint
+    to Ray Train.
+
+    If you want more customized reporting logics, please implement your own
+    callbacks following the Transformers integration user guides.
+
+    Note that users should ensure that the logging, evaluation, and saving frequencies
+    are properly configured to ensure that the monitoring metric is always up-to-date
+    when `transformers.Trainer` saves a checkpoint.
+
+    Suppose the monitoring metric is reported from evaluation stage:
+
+    Some valid configurations:
+        - evaluation_strategy == save_strategy == "epoch"
+        - evaluation_strategy == save_strategy == "steps", save_steps % eval_steps == 0
+
+    Some invalid configurations:
+        - evaluation_strategy != save_strategy
+        - evaluation_strategy == save_strategy == "steps", save_steps % eval_steps != 0
+
+    """
+
+    def on_save(self, args, state, control, **kwargs):
+        """Event called after a checkpoint save."""
+        with TemporaryDirectory() as tmpdir:
+            metrics = state.log_history[-1] if state.log_history else {}
+
+            source_ckpt_path = transformers.trainer.get_last_checkpoint(args.output_dir)
+            target_ckpt_path = os.path.join(tmpdir, "checkpoint")
+            shutil.copytree(source_ckpt_path, target_ckpt_path)
+            checkpoint = Checkpoint.from_directory(tmpdir)
+
+            ray.train.report(metrics=metrics, checkpoint=checkpoint)
+
+
+@PublicAPI(stability="alpha")
+def prepare_trainer(trainer: Trainer) -> Trainer:
+    """Prepare your HuggingFace Transformer Trainer for Ray Train."""
+    base_trainer_class: Type[transformers.trainer.Trainer] = trainer.__class__
+
+    class RayTrainer(base_trainer_class):
```

Comment:
Just in case this shows up somewhere, probably better to be more explicit
```suggestion
    class RayTransformersTrainer(base_trainer_class):
```